docs/backend/backend-overview.md

```diff
@@ -0,0 +1,210 @@
+
+### Small Iterations
+
+While we may cast a vision for the future, we desire to get there via a series
+of small iterations. We have a working backend pipeline already and will
+continue to run it every week. In parallel, we can work to evolve it through a
+series of small iterations that evolve our processes over time to use new
+service-based interfaces, gradually evolving the execution process that occurs
+each week.
+
+We do not want a new effort happening in parallel to what is already working
+today.
+
+### Minimize Mandatory External Dependencies
+
+Major external dependencies should be made optional where feasible to do so.
+The more of these we require, the more complicated it will be to reuse this
+system in other environments.
+
+In other cases, it may be appropriate to provide an abstraction for the
+dependency so that alternatives can be put in place. Alternate Object Storage
+systems are a good example.
+
+### Make Development Accessible
+
+Most people will not have direct access to a large cluster to test changes
+to this system. We should create development environments and minimal
+configurations as much as possible to lower the barrier to doing useful
+development and testing with these components.
+
+### Model Independence
+
+We desire to keep the backend flexible such that serving and training alternate
+models in the future is not more work than absolutely necessary.
+
+## Key Technology Decisions
+
+### Kubernetes
+
+We consider Kubernetes an acceptable dependency for the Backend Service. While
+it may be technically possible to run most of the code elsewhere, the project
+will focus its energy on running this collection of components in a
+Kubernetes-based environment.
+
+This also applies to when these components are run on a single server. In that
+environment, we expect one of the single-node Kubernetes distributions to be
+used. One example is [Microshift](https://github.com/openshift/microshift/), but
+there are others.
+
+Note however that the core backend pipeline components should also be able to
+run directly on a single server without the backend services. These capabilities
+can be executed directly using the `ilab` CLI.
+
+A development environment capable of running within
+[kind](https://kind.sigs.k8s.io/) would also be incredibly convenient, though not
+as a production target environment.
+
+### Programming Languages
+
+Python is the primary language used in the current backend. It is also the
+de facto choice for AI/ML developers. We will continue using Python for core
+functionality, especially when it comes to interfacing with AI/ML libraries like
+PyTorch.
+
+Another significant element of the Backend Service is automating workloads on
+Kubernetes. The most popular language in this context is Go. Kubernetes itself
+is in Go, and the Go client libraries are the most mature.
+
+To best support the full scope of the project, we will use both Python and Go,
```

Comment:
Per our earlier conversation, I think it may be worth including `rust` in the list for service implementations. Alternatively, it may be worth noting that this is not an exclusive list, but instead an indication of supported preference.

Reply:
Yeah, I remember the conversation, but I couldn't come up with a concrete example of something in scope that fit the criteria we discussed for where it would make sense.  I'm open to it, but I just want to provide more concrete justification. I feel like I can provide those for Python and Go.

To be clear I'm focused on InstructLab specific services with these comments on implementation details.

Reply:
Yep, I think limiting to `python` and `go` for the time being is fine as long as it's not phrased as an exclusive list. Maybe we could add a sentence along the lines of:

> Additional languages may be considered for addition to the supported list as needed through additional proposal documents in this repo.

Reply:
I like the above wording by Gabe